### PR TITLE
Adds new package @automattic/calypso-doctor

### PR DESCRIPTION
**Pull request by scinos**
#### Changes proposed in this Pull Request

Adds a new package `@automattic/calypso-doctor`. This package is a tool that performs some tests on the developer's system to verify they have the best settings to develop Calypso smoothly.

It is designed to be modular, adding new tests should be easy. In the future it can be expanded to automatically fix the problems, validate other sets of tests (for example, VSCode settings)...

For now it runs on `postinstall` on the main project, but it doesn't fail the installation in any way. Is up to the developer to manually apply the fixes.

Example of a run:

image

After applying the suggested fixes:

image

#### Testing instructions

Checkout the branch, run `yarn` and see what errors you get. Try to apply the suggested fixes and see if the errors go away.

### CODE REVIEW

File: packages/calypso-doctor/cli.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env node
+const yargs = require( 'yargs' );
+const chalk = require( 'chalk' );
+const { runEvaluations } = require( './index.js' );
+
+const main = async ( args ) => {
+	const results = await runEvaluations( { fix: args.fix } );
```
Comment (saramarcondes) [inline packages/calypso-doctor/cli.js:7]: The argument to `runEvaluations` is unused. Is that intentional or left over from a previous idea?

**Review by sarayourfriend — APPROVED** [review]
I love this! Just a couple of things off the top of my head that might improve the API:

1) Change the signature of `test` to pass a `fail` and `pass` function:

```js
const evaluation = {
	test: ( pass, fail ) => {
		if ( ! process.env.PUPPETEER_SKIP_DOWNLOAD ) {
			return fail( 'PUPPETEER_SKIP_DOWNLOAD is not set' );
		}

		return pass();
	},
};
```

And raise a runtime failure if something other than an object with `result: boolean` is returned.

Mostly this is to cut out some of the boilerplate I'm seeing.

2) Shell constants fixes. All of these should have `export` in front of them. Additionally, I wonder if we couldn't add an `autofix` option that would just `echo ${ expression } >> ${ shell }`. That seems like a v0.2.0 thing to me, but could be a nice additional feature.

3) Another additional feature that could be nice is a concept of "alerts" that can be dismissed. We had this at my previous job and it helped with making sure everyone working in the repository had the most up-to-date information regarding process changes and the like. The dismissals were held locally in a `~/.tool-name` file which was just a JSON file mapping `{ "alert-name": boolean }`.

But overall LGTM, feel free to ship w/o any changes, just some ideas!

File: packages/calypso-doctor/cli.js (hunk 1 above)
Comment (scinos) [reply packages/calypso-doctor/cli.js:7]: Left over from my attempt to implement autofix, I'll clean it up, thanks!

**Comment by scinos** [comment]
@saramarcondes Thanks for the feedback.

Great idea with the `pass`/`fail` API, it makes writing new evaluations much simpler. Also I love the idea of having some sort of "skip this alarm for me, I know what I'm doing", it will prevent a lot of noise. I'll implement that for 0.2.0.

I'm a bit concerned about the autofix. It always scares me to automatically change the developer's system, so many things can go wrong (what if I don't detect the correct file to change? what if I corrupt it somehow?). So I started to implement it but chicken out mid implementation. I decided to emulate `brew` and just let the developer know what we think it should be done, but don't do anything on their behalf.

Let's see how devs use this tool first? If we end up adding a ton of new and more complex evaluations we may need some sort of autofix anyway.

**Pull request merged**